eth/catalyst/api.go

```diff
@@ -682,3 +682,21 @@ func (api *ConsensusAPI) heartbeat() {
 		}
 	}
 }
+
+// GetPayloadBodiesV1 implements engine_getPayloadBodiesV1 which allows for retrieval of a list
+// of block bodies by the engine api.
+func (api *ConsensusAPI) GetPayloadBodiesV1(hashes []common.Hash) ([]*types.Body, error) {
+	var bodies []*types.Body
+	for _, hash := range hashes {
+		body := api.eth.BlockChain().GetBody(hash)
+		if body == nil {
```

Comment:
This doesn't seem to follow the spec? Would need to `continue` when `body == nil` 

> Client software **MUST** skip the payload body in the response array if the data of this body is missing. For instance, if the request is `[A.block_hash, B.block_hash, C.block_hash]` and client software has data of payloads `A` and `C`, but doesn't have data of `B`, the response **MUST** be `[A.body, C.body]`.